solr/src/main/scala/akka/stream/alpakka/solr/SolrFlowStage.scala

```diff
@@ -84,14 +257,12 @@ private[solr] final class SolrFlowLogic[T, C](
     tryPull()
 
   override def onPush(): Unit = {
-    queue.enqueue(grab(in))
+    val messagesIn = grab(in)
+
     state match {
       case Idle => {
         state = Sending
-        val messages = (1 to settings.bufferSize).flatMap { _ =>
-          queue.dequeueFirst(_ => true)
-        }
-        sendBulkToSolr(messages)
+        sendBulkToSolr(messagesIn)
       }
       case _ => ()
```

Comment:
Since the messages are no longer enqueued they will be lost here if the `state` is not `Idle`.

Reply:
Yep, you're right. Did you think we do not need anymore state management? Or something else?

Reply:
I have not looked further, maybe that is already the case, but it this stage should not be pushed when it is in state other than `Idle`. That means, that this stage should only pull, when it is in state `Idle`.

Reply:
I ve chosen the logical option: I do not manage the internal state since there's no more internal batching